android/app/src/main/java/edu/berkeley/eecs/e_mission/DisplayResultSummaryActivity.java

```diff
@@ -1,104 +1,137 @@
-
 import android.annotation.SuppressLint;
 import android.app.Activity;
 import android.content.Context;
-import android.net.http.AndroidHttpClient;
+import android.net.http.HttpResponseCache;
 import android.os.AsyncTask;
 import android.os.Bundle;
+import android.util.Log;
 import android.webkit.WebView;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.message.BasicNameValuePair;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
 import edu.berkeley.eecs.e_mission.auth.GoogleAccountManagerAuth;
 import edu.berkeley.eecs.e_mission.auth.UserProfile;
 
 @SuppressLint("SetJavaScriptEnabled")
 public class DisplayResultSummaryActivity extends Activity {
+    
+    
+    private WebView displaySummaryView;
+    private ClientStatsHelper statsHelper;
+    
+    @Override
+    public void onCreate(Bundle savedInstanceBundle) {
+        super.onCreate(savedInstanceBundle);
+        statsHelper = new ClientStatsHelper(this);
+        setContentView(R.layout.activity_display_result_summary);
+        displaySummaryView = (WebView) findViewById(R.id.displayResultSummaryView);
+        displaySummary();
+    }
+
+
+    void displaySummary() {
+        final long startMs = System.currentTimeMillis();
+        final Context thisContext = this;
+        final String userName = UserProfile.getInstance(this).getUserEmail();
+        final String result_url = AppSettings.getResultUrl(this);
+        
+        AsyncTask<Void, Void, String> task = new AsyncTask<Void, Void, String>() {
+            @Override
+            protected String doInBackground(Void... params) {
+                try {
+                    String userToken = GoogleAccountManagerAuth.getServerToken(thisContext, userName);
+                    // TODO: Restructure this later to combine with the data sync class
+                    URL url = new URL(result_url);
+                    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
 
+                    connection.setUseCaches(true);
+                    connection.setDoOutput(false);
+                    connection.setDoInput(true);
+                    connection.setReadTimeout( 10000 /*milliseconds*/ );
+                    connection.setConnectTimeout(15000 /* milliseconds */);
+                    connection.setRequestMethod("GET");
+                    connection.setRequestProperty("User", "" + userToken);
```

Comment:
I remember that comment now. I don’t know anything about background syncing, but I can read up on it. I imagine I would set some kind of background thread that runs every X minutes and then keep the same code that I currently have. This way we keep the cache updated every X minutes, but don’t actually invalidate the cache for a much longer length of time. 




I also will need to figure out how to make a new HTTP request in the background that updates the cache, but simultaneously does not use the entry currently in the cache.

On Tue, Mar 3, 2015 at 3:58 PM, shankari <notifications@github.com> wrote:

>>  
>> +                    connection.setUseCaches(true);
>> +                    connection.setDoOutput(false);
>> +                    connection.setDoInput(true);
>> +                    connection.setReadTimeout( 10000 /*milliseconds*/ );
>> +                    connection.setConnectTimeout(15000 /* milliseconds */);
>> +                    connection.setRequestMethod("GET");
>> +                    connection.setRequestProperty("User", "" + userToken);
> lowercase user instead of User to be consistent with the post messages?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/e-mission/e-mission-phone/pull/32/files#r25740693